src/metabase/api/util/handlers.clj

```diff
@@ -0,0 +1,66 @@
+(ns metabase.api.util.handlers
+  "TODO -- consider renaming this to [[metabase.api.handlers]]."
+  (:require
+   [clojure.string :as str]
+   [compojure.core :as compojure]
+   [medley.core :as m]
+   [metabase.api.macros :as api.macros]
+   [metabase.api.open-api :as open-api]))
+
+(set! *warn-on-reflection* true)
+
+(defn- split-path [^String path]
+  (when (some-> path (str/starts-with? "/"))
+    (if-let [next-slash-index (str/index-of path "/" 1)]
+      [(subs path 0 next-slash-index) (subs path next-slash-index (count path))]
+      [path "/"])))
+
+(defn- -route-map-handler [route-map]
+  (fn [request respond raise]
+    (if-let [[prefix rest-of-path] (split-path ((some-fn :path-info :uri) request))]
+      (if-let [handler (get route-map prefix)]
+        (let [request' (assoc request :path-info rest-of-path)]
```

Comment:
nit: I thought x' implies x but with a modification. But request' doesn't share that relation with request. What about naming it the-request or something similar

Reply:
How is `request'` which is `(assoc request :path-info rest-of-path)` not `request` with a modification?

Reply:
Oh I misread this. I thought request was one of the async handlers. Nevermind